File: public/PublicMedia/static/solutions/clojure/problem1.clj
(comment
  "Purpose: Find the sum of all the multiples of 3 or 5 below 1000.
   Author: Eben Quenneville
   Date: 2023-03-04")

(defn good? "returns n if it is a multiple of 3 or 5, 0 otherwise" [n]
  (if (== (mod n 3) 0) n (if (== (mod n 5) 0) n 0)))
;; Main Driver code
(defn problem1 []
  (reduce + (map good? (range 1 1000))))

;; Assert the solution
(defn main []
  (assert (== (problem1) 233168))
  (println "Output: " (problem1)))
(main)

File: public/PublicMedia/static/solutions/rust/problem1.rs
/*
Purpose: Find the sum of all the multiples of 3 or 5 below 1000.

Author: Eben Quenneville
Date: 2022-09-03
*/
fn main() {
    let mut sum = 0;
    for i in 1..1000 {
        if i % 3 == 0 {
            sum += i;
            continue;
        }
        if i % 5 == 0 {
            sum += i;
        }
    }
    println!("{} = 233168", sum)
}
